images/mod-tile/start.py

```diff
@@ -19,16 +19,18 @@
 environ['PGPASSWORD'] = environ['POSTGRES_PASSWORD']
 
 
-def run_subprocess_async(command, wait=False):
+def run_subprocess_async(command, stdout, stderr, wait=False):
     """
     Run a subprocess command in a thread
     Args:
         command (string): the command to run as a string
+        stdout (callable): a callable to pass stdout to
+        stderr (callable): a callable to pass stderr to
         wait (bool, optional): whether to wait until thread termination. Defaults to False.
     """
-    thread = run_command_async(command, log.info, log.error,
-        (lambda exit_code: log.info(f'command: {command}, ended with exit code: {exit_code}')),
-        (lambda: log.info(f'command: {command} completed successfully')))
+    thread = run_command_async(command, stdout, stderr,
+        (lambda exit_code: log.info(f'command: {command} - ended with exit code: {exit_code}')),
+        (lambda: log.info(f'command: {command} - completed successfully')))
     if wait:
         thread.join()
```

Comment:
I think `osmeterium` offer great API, without a need to wrap it with another function, the common code you share between the `async` and `sync` versions are the callbacks for `failure` and `successful` command.

Reply:
removed run_subprocess_async function, using run_command_async directly. done